src/parser/core/modules/Swiftcast.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {RotationTable} from 'components/ui/RotationTable'
+import {ActionLink} from 'components/ui/DbLink'
+
+import {BuffWindowModule} from 'parser/core/modules/BuffWindow'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import {SEVERITY} from 'parser/core/modules/Suggestions'
+import {getDataBy} from 'data/getDataBy'
+
+export default class Swiftcast extends BuffWindowModule {
+	static handle = 'swiftcast'
+	static title = t('core.swiftcast.title')`Swiftcasts`
+
+	buffAction = ACTIONS.SWIFTCAST
+	buffStatus = STATUSES.SWIFTCAST
+
+	expectedGCDs = {
+		expectedPerWindow: 1,
+		suggestionContent: <Trans id="core.swiftcast.suggestions.missedgcd.content">
+			Try to use at least one move during every <ActionLink {...ACTIONS.SWIFTCAST} />
+		</Trans>,
+		severityTiers: {
+			1: SEVERITY.MAJOR,
+		},
+	}
+
+	// override to only use gcd casts
+	onCast(event) {
+		const action = getDataBy(ACTIONS, 'id', event.ability.guid)
+
+		if (!action || action.autoAttack || !action.castTime) {
+			return
+		}
+
+		if (this.activeBuffWindow) {
+			this.activeBuffWindow.rotation.push(event)
+		}
+	}
+
+	// sooper hacky...
+	output() {
```

Comment:
It seems weird to me to override this just to chop off the headers of the rotation table.  Especially given that you're still potentially allowing for requiredGCDs or trackedActions... what identifies those columns to the viewer without the header row?  Is this something that could just work with the base class output() function?

Reply:
This actually does keep the headers -- it just removes the required GCDs column. I'm definitely thinking that overriding `output()` is a good solution because the original function is geared at showing how many moves happened during a buff window instead of the singular move that was used with swiftcast. I like the idea that there's a "go to point in timeline" button, and the move that was used with swiftcast - the 'required number of moves' seems hard to use in this context.

Reply:
The default behavior of the BuffWindow module is to only show any given column if the associated class property is overridden, so that it is not null.  I would expect that just using the default output() behavior given the rest of this class, you should see a table that has: timestamp with goTo link, column headed "GCDs" with the actual/expected GCDs used, and the rotation used.

Reply:
Ah, yes, that is definitely the default output. But as a side effect of overriding `onCast` to ignore actions without cast time makes the "GCDs" column bug out a bit and report actual as 1, and expected as 0.

I fully admit that BuffWindow isn't really made for this purpose, but maybe another table format might be more useful anyway?

Reply:
Ah, the 1/0 is being caused by reduceExpectedGCDsEndOfFight().  Since swiftcast pops early when used, you can just override that to always return 0 and that will fix that issue.

Reply:
Oh, cool. I'll un-override output and put in a patch for that, then. Still, what about a better way to change the column 'Rotation' to 'Ability Used', or something like that?

Reply:
I pushed up those changes, and things definitely look better for the one edge case I was experiencing.

![](https://i.imgur.com/ZnKsi6I.png)